### PR TITLE
error check subaccount and make private_key required

### PR DESCRIPTION
**Pull request by wakamex**

### CODE REVIEW

File: hundred_x/client.py [hunk 1]
```diff
@@ -32,13 +32,15 @@
...
         self,
+        private_key: str,
         env: Environment = Environment.TESTNET,
-        private_key: str = None,
         subaccount_id: int = 1,
     ):
         """
         Initialize the client with the given environment.
         """
+        if subaccount_id < 0 or subaccount_id > 255:
```
Comment (8ball030) [inline hundred_x/client.py]: Great catch

File: tests/test_client.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def setUp(self):
         """
         Set up the Client class tests.
         """
-        self.client = HundredXClient(env=self.environment, private_key=TEST_PRIVATE_KEY, subaccount_id=1)
+        self.client = HundredXClient(private_key=TEST_PRIVATE_KEY, env=self.environment, subaccount_id=1)
```
Comment (8ball030) [inline tests/test_client.py:67]: Please see above :)

File: hundred_x/client.py [hunk 3]
```diff
@@ -32,13 +32,15 @@
 class HundredXClient:
     def __init__(
         self,
+        private_key: str,
         env: Environment = Environment.TESTNET,
-        private_key: str = None,
```
Comment (8ball030) [inline hundred_x/client.py:36]: Why enforce this?

There are unauthenticated endpoints.

A better solution would be the introduction of exceptions,

And then specifically raising the error to the user,

I do agree that we should 100% have some sort of enforcement of authenticated enpoints.

What about using an attribute on the client to indicate which endpoints are authenticated and raising if the private key is none? 

Would be happy to work on this with you?

File: hundred_x/client.py [hunk 4]
```diff
@@ -32,13 +32,15 @@
...
         subaccount_id: int = 1,
     ):
         """
         Initialize the client with the given environment.
         """
+        if 0 <= subaccount_id <= 255:
```
Comment (8ball030) [inline hundred_x/client.py:42]:
```
In [14]: 0 <= -1 <= 255
Out[14]: False

In [15]: 0 <= 1 <= 255
Out[15]: True

In [16]: 0 <= 0 <= 255
Out[16]: True

In [17]: 0 <= 255 <= 255
Out[17]: True

In [18]: 0 <= 256 <= 255
Out[18]: False
```

Comment (8ball030) [reply hundred_x/client.py:42]: Great catch btw @wakamex

File: hundred_x/client.py (hunk 3 above)
Comment (wakamex) [reply hundred_x/client.py:36]: I didn't even think of that. should be easy throw errors on functions that need private key.

**Comment by wakamex** [comment]
closed in favor of https://github.com/8ball030/hundred_x/pull/16

**Pull request closed without merging**